sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py

```diff
@@ -211,14 +211,17 @@ def from_connection_string(
 
     async def send_batch(
         self,
-        event_data_batch: EventDataBatch,
+        event_data_batch: Union[EventDataBatch, List[EventData]],
         *,
+        partition_id: Optional[str] = None,
```

Comment:
What happens if a user pass in an `EventDataBatch` and passes a different `partition_id` value in the call?

Reply:
partition_key is a part of the EventDataBatch. Once an EventDataBatch is built, the partition_key can't be changed for two reasons:
1. The EventDataBatch size will be changed if partition_key is changed.
2. We serialize event(with partition_key) into EventDataBatch when the event is added to EventDataBatch. It's hard to change it once serialized.

To make it easy for users to understand, when the input is an EventDataBatch, we'll validate that partition_key and partition_id must be None. We can raise a TypeError.

Reply:
I wonder if it would help if we moved `partition_key` and `partition_id` in kwargs. Even in the async version.
That would be less distracting.